docs/topics/jvm/java-to-kotlin-interop.md

```diff
@@ -335,20 +336,25 @@ public class BB8 implements Robot {
 
 ### Compatibility mode for default methods
 
-If there are clients that use your Kotlin interfaces compiled without the new `-Xjvm-default=all` option, then they can
+If there are clients that use your Kotlin interfaces compiled without the `-Xjvm-default=all` option, then they can
 be incompatible with the same code compiled with this option. 
 
-To avoid breaking the compatibility with such clients, compile your Kotlin code in the _compatibility mode_ by specifying
-the `-Xjvm-default=all-compatibility` compiler option. In this case, all the code that uses the previous version will 
- work fine with the new one. However, the compatibility mode adds some overhead to the resulting bytecode size.
+Before Kotlin 1.6.20 to avoid this compatibility issue, 
+[this blog post](https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/#JvmDefaultWithoutCompatibility) 
+recommended using the `-Xjvm-default=all-compatibility` mode and the `@JvmDefaultWithoutCompatibility` annotation 
+for interfaces that didn't need such compatibility. This approach had some disadvantages:
+* It was easy to forget to add the annotation when a new interface was added.
+* Usually there are more interfaces in non-public parts than in the public API, so a developer was ending up having this annotation in many places in their code.
+
+Now, you can use the `-Xjvm-default=all` mode and mark interfaces with the `@JvmDefaultWithCompatibility` annotation. 
```

Comment:
Just a reminder for us: to provide the link to the @JvmDefaultWithCompatibility' description here https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/ when it's ready.